jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java

```diff
@@ -91,7 +91,7 @@ public void deconfigure(WebAppContext context) throws Exception
     {
         //if it wasn't explicitly configured by the user, then unset it
         Boolean tmpdirConfigured = (Boolean)context.getAttribute(TEMPDIR_CONFIGURED);
-        if (tmpdirConfigured != null && !tmpdirConfigured)
+        if (tmpdirConfigured == null || !tmpdirConfigured)
             context.setTempDirectory(null);
```

Comment:
I would prefer:
```
        if (!(context.getAttribute(TEMPDIR_CONFIGURED) instanceof Boolean tmpdirConfigured && tmpdirConfigured))
            context.setTempDirectory(null);
```

as this avoids a chance of a caste error, the caste itself, a null check and a variable.

Reply:
I think it's more confusing, but ok.